docs/examples/spatial_transformation/histogram_matching.jl

```diff
@@ -0,0 +1,78 @@
+# ---
+# cover: assets/chelsea.png
+# title: Histogram Matching
+# author: Ashwani Rathee
+# date: 2020-11-21
+# ---
+
+# This demo demonstrates the feature of histogram matching.
+
+# In histogram matching,the pixels of source image is manipulated so that the histogram 
+# of reference image matches with source image's histogram.
+
+# Here,histogram matching is shown for gray scale image.But,it can also be applied on 
+# RGB images too.Histogram matching can be used to normalize two images, when the images 
+# were acquired at the same local illumination (such as shadows) over the same location,
+# but by different sensors,atmospheric conditions or global illumination. 
+
+# Importing packages
+
+using ImageContrastAdjustment
+using Images
+using Plots
+using TestImages
+
+# Load example images: one source image and one reference image
+
+img_source = testimage("chelsea");
+img_reference = testimage("coffee");
+
+# Applying histogram matching on source image using reference image
+
+img_transformed=adjust_histogram(img_source, Matching(targetimg = img_reference));
+
+# **Results** : 
+
+mosaicview(img_source,img_reference,img_transformed;nrow=1)
+save("assets/chelsea.png",img_source) #src
```

Comment:
I hope we can keep some more meaningful results as our cover here so that potential readers immediately get what "Histogram Matching" is. Perhaps a gif that shows the matching progress would be good enough?

Reply:
Sure :)

Reply:
@johnnychen94 I have cleaned up the spaces and stops part of the demo with grammarly and formatted it with vs code.Just learned to create gif with save with help of @timholy from here :https://discourse.julialang.org/t/plotting-a-sequence-of-images-as-a-gif/23808/25
By creating a 3d array of images
```
using FileIO, ImageMagick, Colors, FixedPointNumbers
using TestImages
img1=testimage("cameraman")
img2=testimage("fabio_color_512")
img3=testimage("lena_color_512")
save("/home/ashwani/test.gif", cat(img1,img2,img3;dims=3))
```
But issue is that, whenever dimension mismatch happens it shows error....so I think what I could do is that I create a gif of initial chelsea and final chelsea as their size will remain same

Reply:
Maybe use `PaddedViews` to make the sizes equal:

```julia
cat(sym_paddedviews(colorant"white", img1,img2,img3)...; dims=3)
```

Also, it should be better to use `ImageMagick.save` explicitly since other image IO backends don't support `gif` saving yet. Although not documented, `ImageMagick.save` also accepts a `fps` keyword.

An live example for this can be found in https://github.com/JuliaImages/juliaimages.github.io/blob/4749786007ffd4553ca265bd6115126e3a1047b0/docs/examples/color_channels/color_separations_svd.jl#L51